server/src/main/java/org/opensearch/action/admin/indices/get/GetIndexResponse.java

```diff
@@ -301,10 +300,8 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                     } else {
                         MappingMetadata mappings = null;
                         for (final ObjectObjectCursor<String, MappingMetadata> typeEntry : indexMappings) {
```

Comment:
Probably future change, but I believe we would significantly reduce the amount of "heap garbage" when replace `ImmutableOpenMap<String, MappingMetadata>` with just `MappingMetadata` since there is only one mapping from now on.

Reply:
Yeah, I think that would be a good followup PR.